[GRPC] Add documentation for GRPC plugin, and Bulk+Search GRPC APIs (#9649)
opensearch-project/documentation-website: _api-reference/index.md

- [Transforms API]({{site.url}}{{site.baseurl}}/im-plugin/index-transforms/transforms-apis/)
- [Hot reload TLS certificates]({{site.url}}{{site.baseurl}}/security/configuration/tls/#hot-reloading-tls-certificates)

## gRPC APIs
**Introduced 3.0**
{: .label .label-purple }

> OpenSearch 3.0 introduces gRPC APIs, offering a high-performance alternative to traditional REST interfaces. These APIs use the gRPC protocol to provide more efficient communication with OpenSearch clusters. For more information and supported APIs, see [gRPC APIs]({{site.url}}{{site.baseurl}}/api-reference/grpc-apis/).

Reviewer: "introduces" => "introduced"?

Reviewer 2: Reworded.

> OpenSearch 3.0 introduces gRPC APIs, offering a high-performance alternative to traditional REST interfaces. These APIs use the gRPC protocol to provide more efficient communication with OpenSearch clusters. For more information and supported APIs, see [gRPC APIs]({{site.url}}{{site.baseurl}}/api-reference/grpc-apis/).

Reviewer 2: ```suggestion
Starting with OpenSearch 3.0, you can use gRPC APIs---a high-performance alternative to traditional REST interfaces. These APIs use the gRPC protocol to provide more efficient communication with OpenSearch clusters. For more information and supported APIs, see [gRPC APIs]({{site.url}}{{site.baseurl}}/api-reference/grpc-apis/).
```

- [Transforms API]({{site.url}}{{site.baseurl}}/im-plugin/index-transforms/transforms-apis/)
- [Hot reload TLS certificates]({{site.url}}{{site.baseurl}}/security/configuration/tls/#hot-reloading-tls-certificates)

> ## GRPC API reference

Reviewer 2: No need for a dedicated section. You can just list the section in the list above. We should remove "introduced 1.0" from the API reference index page because not every single one was introduced in 1.0. Thanks.

Author: Since GRPC is a completely different protocol from HTTP/REST, I was wondering if we should rename from "REST API reference" to something else?  e.g. just "API reference"?
